src/Mod/Cloud/App/AppCloud.cpp

```diff
@@ -777,13 +1078,21 @@ bool Cloud::CloudWriter::shouldWrite(const std::string& , const Base::Persistenc
 void Cloud::CloudWriter::pushCloud(const char *FileName, const char *data, long size)
 {
         struct Cloud::AmzData *RequestData;
+        struct Cloud::AmzDatav4 *RequestDatav4;
         CURL *curl;
         CURLcode res;
 	struct data_buffer curl_buffer;
 
 	char path[1024];
         sprintf(path, "/%s/%s", this->Bucket, FileName);
-        RequestData = Cloud::ComputeDigestAmzS3v2("PUT", "application/octet-stream", path, this->TokenSecret, data, size);
+
+        std::string strURL(this->URL);
+        eraseSubStr(strURL,"http://");
+        eraseSubStr(strURL,"https://");
+	if ( this->ProtocolVersion == "2" )
+	        RequestData = Cloud::ComputeDigestAmzS3v2("PUT", "application/octet-stream", path, this->TokenSecret, data, size);
+	else
+		RequestDatav4 = Cloud::ComputeDigestAmzS3v4("PUT", strURL.c_str(), "application/octet-stream", path, this->TokenSecret, data, size, NULL, this->Region);

```

Comment:
It seems there are many places in this pull request that mix tabs and spaces. Maybe you'd like to correct this.

Reply:
Shall be done in new commit